internal/clients/http/sources/sources_client.go

```diff
@@ -67,38 +72,41 @@ type dataElement struct {
 func (c *SourcesClient) Ready(ctx context.Context) error {
 	resp, err := c.client.ListApplicationTypes(ctx, &ListApplicationTypesParams{}, headers.AddSourcesIdentityHeader)
 	if err != nil {
-		ctxval.Logger(ctx).Error().Err(err).Msgf("Readiness request failed for sources: %s", err.Error())
+		c.logger.Error().Err(err).Msgf("Readiness request failed for sources: %s", err.Error())
 		return err
 	}
 	defer resp.Body.Close()
-	if !parsing.IsHTTPStatus2xx(resp.StatusCode) {
-		ctxval.Logger(ctx).Warn().Msgf("Readiness response from sources: %d", resp.StatusCode)
-		return SourcesClientErr
+
+	err = http.HandleHTTPResponses(ctx, resp.StatusCode)
+	if err != nil {
+		return fmt.Errorf("ready call: %w", err)
 	}
 	return nil
 }
 
 func (c *SourcesClient) ListProvisioningSources(ctx context.Context) (*[]clients.Source, error) {
-	ctxval.Logger(ctx).Info().Msg("Listing provisioning sources")
+	c.logger.Trace().Msg("Listing provisioning sources")
+
 	appTypeId, err := c.GetProvisioningTypeId(ctx)
 	if err != nil {
-		ctxval.Logger(ctx).Warn().Err(err).Msg("Failed to get provisioning type id")
+		c.logger.Warn().Err(err).Msg("Failed to get provisioning type id")
 		return nil, fmt.Errorf("failed to get provisioning app type: %w", err)
 	}
+
 	resp, err := c.client.ListApplicationTypeSourcesWithResponse(ctx, appTypeId, &ListApplicationTypeSourcesParams{}, headers.AddSourcesIdentityHeader)
 	if err != nil {
-		ctxval.Logger(ctx).Warn().Err(err).Msg("Failed to fetch ApplicationTypes from sources")
+		c.logger.Warn().Err(err).Msg("Failed to fetch ApplicationTypes from sources")
 		return nil, fmt.Errorf("failed to get ApplicationTypes: %w", err)
 	}
-	statusCode := resp.StatusCode()
 
-	if parsing.IsHTTPNotFound(statusCode) {
-		return nil, NewNotFoundError(ctx, "sources not found")
-	}
-	if !parsing.IsHTTPStatus2xx(statusCode) {
-		ctxval.Logger(ctx).Warn().Msgf("Sources replied with unexpected status while fetching sources of application type: %v", statusCode)
-		return nil, SourcesClientErr
+	err = http.HandleHTTPResponses(ctx, resp.StatusCode())
+	if err != nil {
+		if errors.Is(err, clients.NotFoundErr) {
+			return nil, fmt.Errorf("list provisioning sources call: %w", SourceNotFoundErr)
+		}
+		return nil, fmt.Errorf("list provisioning sources call: %w", err)
```

Comment:
IMHO this looks better:

```suggestion
		return nil, fmt.Errorf("list provisioning sources call: %w", err)
```